src/sentry/api/endpoints/group_similar_issues_embeddings.py

```diff
@@ -49,13 +57,45 @@ def get_stacktrace_string(exception: Mapping[Any, Any], event: GroupEvent) -> st
     return "\n".join(output)
 
 
+class FormattedSimilarIssuesEmbeddingsData(TypedDict, total=False):
+    exception: float
+    message: float
+    shouldBeGrouped: str
+
+
 @region_silo_endpoint
 class GroupSimilarIssuesEmbeddingsEndpoint(GroupEndpoint):
     owner = ApiOwner.ISSUES
     publish_status = {
         "GET": ApiPublishStatus.PRIVATE,
     }
 
+    def get_formatted_results(
+        self, responses: Sequence[Optional[SimilarIssuesEmbeddingsData]], user: User | AnonymousUser
+    ) -> Sequence[Tuple[Mapping[str, Any], Mapping[str, Any]]]:
+        """Format the responses using to be used by the frontend."""
+        group_data = {}
+        for response in responses:
+            if response:
+                formatted_response: FormattedSimilarIssuesEmbeddingsData = {}
+                formatted_response.update({"message": response["message_similarity"]})
+                formatted_response.update({"exception": response["stacktrace_similarity"]})
+                formatted_response.update(
+                    {"shouldBeGrouped": "Yes" if response["should_group"] else "No"}
+                )
+                group_data.update({response["parent_group_id"]: formatted_response})
+
+        serialized_groups = {
+            int(g["id"]): g
+            for g in serialize(
+                list(Group.objects.get_many_from_cache(group_data.keys())), user=user
+            )
+        }
+
+        return [
+            (serialized_groups[group_id], group_data[group_id]) for group_id in group_data.keys()
```

Comment:
Do you need to do `group_data.keys()`? Would `group_id in group_data` do the same thing?

Reply:
Thanks for both your suggestions!